### ISSUE TITLE
Tooltip: for accessibility provide multiple showEvent as default mouseover, focus and touchstart and ability to close by Esc keypress

### ISSUE THREAD

**Issue by sunriseO** [primefaces/primefaces#13521]
### Description

If, in addition to hover, you also want to have the tooltip for the focus, for example, you have to define a second p:tooltip with the identical text for the same target. Disadvantage: the aria-describedby attribute of the target element then already contains two IDs for both target elements and reads out the same tooltip text twice. The same phenomenon occurs if you create a third p:tooltip for showEvent=“touchstart”. The IDs for the aria-describedby attribute of the target element are collected with each additional p:tooltip for the same target element and all texts with identical content are read out repeatedly.

In addition, the tooltip cannot be closed by pressing the Esc key. Both aspects needed for accessibility reasons.

### Describe the solution you would like

- default showEvent: mouseover focus touchstart (currently only default mouseover event displaying the tooltip)
  -> no need to define multiple p:tooltip for different showEvent having indentical value to be shown readed repeated by screenreader
- dito default hideEvent: mouseout blur touchend
- Esc keypress should close the tooltip for the target element by default

**Comment by melloware** [comment]
@sunriseO these are great suggestions.  PrimeReact doesn't quite  have the same as this but does have the ESC to close.  Let me do some testing.

**Comment by melloware** [comment]
@sunriseO here is a MonkeyPatch you can use right now to do ...


```js
if (PrimeFaces.widget.Tooltip) {

  PrimeFaces.widget.Tooltip.prototype.init = function(cfg) {
    PrimeFaces.widget.BaseWidget.prototype.init.call(this, cfg);

    this.cfg.showEvent = this.cfg.showEvent ? this.cfg.showEvent.split(' ').map(e => e + '.tooltip').join(' ') : 'mouseenter.tooltip';
    this.cfg.hideEvent = this.cfg.hideEvent ? this.cfg.hideEvent.split(' ').map(e => e + '.tooltip').join(' ') : 'mouseleave.tooltip';
    this.cfg.showEffect = this.cfg.showEffect ? this.cfg.showEffect : 'fade';
    this.cfg.hideEffect = this.cfg.hideEffect ? this.cfg.hideEffect : 'fade';
    this.cfg.showDelay = PrimeFaces.utils.defaultNumeric(this.cfg.showDelay, 400);
    this.cfg.hideDelay = PrimeFaces.utils.defaultNumeric(this.cfg.hideDelay, 0);
    this.cfg.hideEffectDuration = this.cfg.target ? 250 : 1;
    this.cfg.position = this.cfg.position || 'right';
    this.cfg.escape = (this.cfg.escape === undefined) ? true : this.cfg.escape;
    this.cfg.autoHide = (this.cfg.autoHide === undefined) ? true : this.cfg.autoHide;
    this.allowHide = true;

    if (this.cfg.target)
      this.bindTarget();
    else
      this.bindGlobal();
  }
}
```

**Comment by sunriseO** [comment]
the MonkeyPatch works - thank you very much

**Comment by melloware** [comment]
awesome and the Escape key will also be fixed in 15.0.2
